compiler/src/iree/compiler/Dialect/HAL/IR/HALOps.td

```diff
@@ -122,6 +122,7 @@ def HAL_TensorImportOp : HAL_PureOp<"tensor.import", [
     AnyTypeOf<[HAL_Buffer, HAL_BufferView]>:$source,
     TypeAttr:$target_encoding,
     HAL_ShapeDynamicDims:$target_dims,
+    UnitAttr:$consume,
```

Comment:
Triviality - this could be `UnitProp` if we'd like - that'll just let it be a boolean stored next to the op